pymc3/tuning/starting.py

```diff
@@ -69,62 +73,46 @@ def find_MAP(start=None, vars=None, fmin=None,
     except AttributeError:
         gradient_avail = False
 
-    if disc_vars or not gradient_avail :
+    if disc_vars or not gradient_avail:
         pm._log.warning("Warning: gradient not available." +
                         "(E.g. vars contains discrete variables). MAP " +
                         "estimates may not be accurate for the default " +
                         "parameters. Defaulting to non-gradient minimization " +
-                        "fmin_powell.")
-        fmin = optimize.fmin_powell
+                        "'Powell'.")
+        method = "Powell"
 
-    if fmin is None:
+    if method is None:
         if disc_vars:
-            fmin = optimize.fmin_powell
+            method = "Powell"
         else:
-            fmin = optimize.fmin_bfgs
+            method = "BFGS"
 
     allinmodel(vars, model)
 
     start = Point(start, model=model)
     bij = DictToArrayBijection(ArrayOrdering(vars), start)
+    logp_func = bij.mapf(model.fastlogp)
+    x0 = bij.map(start)
 
-    logp = bij.mapf(model.fastlogp)
-    def logp_o(point):
-        return nan_to_high(-logp(point))
-
-    # Check to see if minimization function actually uses the gradient
-    if 'fprime' in getargspec(fmin).args:
-        dlogp = bij.mapf(model.fastdlogp(vars))
-        def grad_logp_o(point):
-            return nan_to_num(-dlogp(point))
-
-        if live_disp and callback is None:
-            callback = Monitor(bij, logp_o, model, grad_logp_o)
-
-        r = fmin(logp_o, bij.map(start), fprime=grad_logp_o, callback=callback, *args, **kwargs)
+    if method in ["CG", "BFGS", "Newton-CG", "L-BFGS-B", "TNC",
+                  "SLSQP", "dogleg", "trust-ncg"]:
```

Comment:
oh good catch, it would be nice to be able to provide a custom method.  I'll try and fix that.